fluent-syntax/src/ast.ts

```diff
@@ -6,6 +6,8 @@
  *
  */
 export abstract class BaseNode {
+  abstract type: string
```

Comment:
The immediate reason for adding this is to allow for `node.type` to be known to be a `string` in template expressions. While this typing is nominally new, it matches all existing practice at least within thie monorepo.

I'm not sure if this should techinically count as a breaking change, but as long as the next version of `@fluent/syntax` is published as 0.18.0, all should be fine even from a semver point of view.

Reply:
Thought: I'd feel comfortable with a minor semver bump.

Reply:
Heh, we're still in 0.x land so a minor and a major bump are effectively the same thing. :)